tools/codegen/dest/register_dispatch_key.py

```diff
@@ -85,6 +98,65 @@ def __call__(self, f: Union[NativeFunctionsGroup, NativeFunction]) -> List[str]:
         else:
             assert_never(f)
 
+    def gets_generated_out_inplace_wrapper(self, f: NativeFunction, g: NativeFunctionsGroup) -> bool:
+        return f.func.kind() is not SchemaKind.functional \
+            and not self.backend_index.has_backend(f) and self.backend_index.has_backend(g.functional)
+
+    def wrapper_kernel_sig(self, f: NativeFunction) -> Union[NativeSignature, DispatcherSignature]:
+        # The prefix is just to ensure uniqueness. The Dispatcher API doesn't guarantee unique kernel names.
+        return kernel_signature(f, self.backend_index, prefix=f'wrapper_{f.func.name.overload_name}_')
+
+    def gen_inplace_out_wrapper(self, f: NativeFunction, g: NativeFunctionsGroup) -> Optional[str]:
+        # Only anonymous definitions get "special treatment" for out/inplace wrappers.
+        # All other functionality can be directly pulled from gen_unstructured.
+        is_generated_wrapper = self.gets_generated_out_inplace_wrapper(f, g)
+        if self.target is not Target.ANONYMOUS_DEFINITION:
+            return self.gen_unstructured(f, is_generated_wrapper=is_generated_wrapper)
+
+        if f.func.kind() is SchemaKind.functional:
+            # Wrappers are generated for out/inplace kernels, using the functional kernel,
+            # so the functional kernel itself is generated normally.
+            return self.gen_unstructured(f)
+        if self.backend_index.has_backend(f):
+            # If the backend provided their own out/inplace kernel, use it.
+            return self.gen_unstructured(f)
+
+        if not is_generated_wrapper:
+            return None
+
+        # Special out/inplace wrapper logic starts here.
+        sig = self.wrapper_kernel_sig(f)
+        name = sig.name()
+
+        # See Note [External Backends Follow Dispatcher convention]
+        jit_args = dispatcher.jit_arguments(f.func)
+        tensors = [a for a in jit_args if isinstance(a, Argument) and a.type == BaseType(BaseTy.Tensor)]
+        print_args_str = ''.join([f' << " {a.name}=" << {a.name}.toString()' for a in tensors])
+
+        functional_result_name = f'{name}_tmp'
+        return_names = cpp.return_names(f)
+        if len(return_names) > 1:
+            updates = '\n  '.join(
+                f'at::_copy_from_and_resize(std::get<{i}>({functional_result_name}), {ret_name});'
```

Comment:
Oh, good catch. I think that any size mismatch errors would be caught in the meta function, so we should never end up in that situation. But getting the codegen to explicitly call a non-resizing function for inplace wrappers definitely expresses intent more clearly.

Reply:
Actually yeah that was a straight up bug- since a lot of XLA kernels don't do shape checks (LazyTensor `meta()` support can't come soon enough). So before when I was using `_copy_from_and_resize()` for inplace ops, we would just re-size `self` on XLA, while CPU would correctly error in the meta function.

using `_copy_from` gives you a pretty crappy error message, but there's probably no point trying to make that better until we start using meta functions.